src/node_watchdog.cc

```diff
@@ -107,11 +115,125 @@ SigintWatchdog::~SigintWatchdog() {
 }
 
 
-void SigintWatchdog::HandleSigint() {
+SignalPropagation SigintWatchdog::HandleSigint() {
   *received_signal_ = true;
   isolate_->TerminateExecution();
+  return SignalPropagation::StopPropagation;
+}
+
+
+void JsSigintWatchdog::Init(Environment* env, Local<Object> target) {
+  Local<FunctionTemplate> constructor = env->NewFunctionTemplate(New);
+  constructor->InstanceTemplate()->SetInternalFieldCount(1);
+  Local<v8::String> js_sigint_watch_dog =
+        FIXED_ONE_BYTE_STRING(env->isolate(), "JsSigintWatchdog");
+  constructor->SetClassName(js_sigint_watch_dog);
+  constructor->Inherit(HandleWrap::GetConstructorTemplate(env));
+
+  env->SetProtoMethod(constructor, "start", Start);
+  env->SetProtoMethod(constructor, "stop", Stop);
+
+  target->Set(env->context(), js_sigint_watch_dog,
+              constructor->GetFunction(env->context()).ToLocalChecked())
+              .Check();
+
+#define SET_SIGNAL_STATE(name, it) \
+  { \
+    Local<v8::String> sig_flag_name = \
+          FIXED_ONE_BYTE_STRING(env->isolate(), #name); \
+    Local<v8::Number> sig_flag_val = v8::Number::New(env->isolate(), it); \
+    target->Set(env->context(), sig_flag_name, sig_flag_val) \
+      .Check(); \
```

Comment:
Style nit: can you line up the backslashes? We usually (but not always) do that at the 79th column. `git grep '\\$' src/` for examples. :-)

Reply:
I just noticed the rule `AlignEscapedNewlines` in file `/.clang-format`. It seems like we don't enforce the rules in `/.clang-format`?

Reply:
The `clang-format` rules affect only `make format-cpp`, e.g. you can use `CLANG_FORMAT_START=master make format-cpp` for autoformatting.